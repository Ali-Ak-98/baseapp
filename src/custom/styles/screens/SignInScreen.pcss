.pg-sign-in-screen {
  display: flex;
  justify-content: center;
  height: calc(100vh - 58px);
  margin-top: 50px;
  background-color: var(--accent);

  &__background {
    background-image: url(assets/images/landing/looksComplex/RocketEngine.svg);
    background-position: right 23%;
    background-repeat: no-repeat;
    background-size: cover;
    content: "";
    display: block;
    height: calc(100% - 60px);
    position: absolute;
    transform: translate3d(0,0,0);
    right: 0%;
    width: 1050px;
    z-index: 1;
    opacity: 0.2;
    overflow: hidden;
  }


  .pg-sign-in-screen__container {
    background: transparent;
    margin-top: 0;
    width: 100%;
    z-index: 2;
  }

  &__body {
    display: flex;
    flex-direction: row-reverse;
    justify-content: center;
    max-width: 1140px;
    margin: 0 auto;
    margin-top: 50px;
  }

  .cr-sign-in-form {
    background: var(--base-bright);
    margin: 0 auto;
    border-radius: calc(var(--gap) * 0.66);
    min-height: calc(var(--big-gap) * 10);

    &__options-group {
      width: 100%;
      height: calc(var(--big-gap) * 0.97);
      border-bottom: 2px solid var(--sign-in-text);
    }


    &__group {
      margin-bottom: 24px;
    }

    &__option {
      background: var(--header);
      .__selected {
        border-bottom: 2px solid var(--sign-in-accent);
        color: var(--color-white);
      }
    }
  }
}

@media (max-width: 768px) {
  .pg-sign-in-screen {
    height: 100% !important;

    &__body {
      flex-direction: column-reverse;
      margin-top: 55px;
    }

    &__container {
      margin: 0 !important;
      padding: 10px;
    }
  }
}
