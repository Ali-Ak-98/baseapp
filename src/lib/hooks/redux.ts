import { useSelector, shallowEqual } from 'react-redux';

import { ReduxState } from 'src/redux/interfaces';
import { LocalizationContainer } from 'src/units/localization/interfaces';

export function useReduxSelector<T>(selector: (state: ReduxState) => T): T {
    return useSelector(selector, shallowEqual);
}

export function useLocalization(): LocalizationContainer {
    return useReduxSelector((x) => x.localization.text);
}
