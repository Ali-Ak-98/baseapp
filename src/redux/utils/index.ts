export * from './action';
export * from './saga-handler';
export * from './wrapper';
export * from './monolite';
