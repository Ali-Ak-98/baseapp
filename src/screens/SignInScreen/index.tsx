import cx from 'classnames';
import * as React from 'react';
import {InjectedIntlProps, injectIntl} from 'react-intl';
import {connect, MapDispatchToPropsFunction, MapStateToProps} from 'react-redux';
import {RouterProps} from 'react-router';
import {withRouter} from 'react-router-dom';
import {SignInComponent, TwoFactorAuth} from '../../components';
import {API, RequestOptions} from '../../api';
import {EMAIL_REGEX, ERROR_EMPTY_PASSWORD, ERROR_INVALID_EMAIL, setDocumentTitle} from '../../helpers';
import {
    RootState,
    selectAlertState, selectShow2faActions,
    selectSignInRequire2FA,
    selectSignUpRequireVerification,
    selectUserFetching,
    selectUserLoggedIn,
    signIn,
    signInError,
    signInRequire2FA,
    show2FAActions,
    signUpRequireVerification,
    alertPush,
} from '../../modules';


interface ReduxProps {
    isLoggedIn: boolean;
    loading?: boolean;
    require2FA?: boolean;
    show2faActions?: boolean;
    requireEmailVerification?: boolean;
}

interface DispatchProps {
    signIn: typeof signIn;
    signInError: typeof signInError;
    signInRequire2FA: typeof signInRequire2FA;
    signInShow2FActions: typeof show2FAActions;
    signUpRequireVerification: typeof signUpRequireVerification;
    alertSuccess: typeof alertPush;
}

interface SignInState {
    email: string;
    emailError: string;
    emailFocused: boolean;
    password: string;
    passwordError: string;
    passwordFocused: boolean;
    otpCode: string;
    error2fa: string;
    codeFocused: boolean;
}

type Props = ReduxProps & DispatchProps & RouterProps & InjectedIntlProps;

class SignIn extends React.Component<Props, SignInState> {
    public state = {
        email: '',
        emailError: '',
        emailFocused: false,
        password: '',
        passwordError: '',
        passwordFocused: false,
        otpCode: '',
        error2fa: '',
        codeFocused: false,
    };

    public componentDidMount() {
        setDocumentTitle('Sign In');
        this.props.signInError({code: undefined, message: undefined});
        this.props.signUpRequireVerification({requireVerification: false});
    }

    public componentWillReceiveProps(props: Props) {
        if (props.isLoggedIn) {
            this.props.history.push('/wallets');
        }
        if (props.requireEmailVerification) {
            props.history.push('/email-verification', {email: this.state.email});
        }
    }

    public render() {
        const {loading, require2FA} = this.props;
        const className = cx('pg-sign-in-screen__container', {loading});
        return (
            <div className="pg-sign-in-screen">
                <div className={className}>{require2FA ? this.render2FA() : this.renderSignInForm()}</div>
            </div>
        );
    }

    private renderSignInForm = () => {
        const {loading} = this.props;
        const {email, emailError, emailFocused, password, passwordError, passwordFocused} = this.state;

        return (
            <SignInComponent
                email={email}
                emailError={emailError}
                emailFocused={emailFocused}
                emailPlaceholder={this.props.intl.formatMessage({id: 'page.header.signIn.email'})}
                password={password}
                passwordError={passwordError}
                passwordFocused={passwordFocused}
                passwordPlaceholder={this.props.intl.formatMessage({id: 'page.header.signIn.password'})}
                labelSignIn={this.props.intl.formatMessage({id: 'page.header.signIn'})}
                labelSignUp={this.props.intl.formatMessage({id: 'page.header.signUp'})}
                emailLabel={this.props.intl.formatMessage({id: 'page.header.signIn.email'})}
                passwordLabel={this.props.intl.formatMessage({id: 'page.header.signIn.password'})}
                receiveConfirmationLabel={this.props.intl.formatMessage({id: 'page.header.signIn.receiveConfirmation'})}
                forgotPasswordLabel={this.props.intl.formatMessage({id: 'page.header.signIn.forgotPassword'})}
                isLoading={loading}
                onForgotPassword={this.forgotPassword}
                onSignUp={this.handleSignUp}
                onSignIn={this.handleSignIn}
                handleChangeFocusField={this.handleFieldFocus}
                isFormValid={this.validateForm}
                refreshError={this.refreshError}
                changeEmail={this.handleChangeEmailValue}
                changePassword={this.handleChangePasswordValue}
            />
        );
    };

    private render2FA = () => {
        const {loading, show2faActions} = this.props;
        const {otpCode, error2fa, codeFocused} = this.state;
        return (
            <TwoFactorAuth
                isLoading={loading}
                show2faActions={show2faActions}
                show2faActionsMsg={this.props.intl.formatMessage({id: '2fa.action.msg'})}
                show2faActionsDisableLabel={this.props.intl.formatMessage({id: '2fa.action.btn.disable'})}
                onSubmit={this.handle2FASignIn}
                onDisable2FA={this.handleDisable2FA}
                title={this.props.intl.formatMessage({id: 'page.password2fa'})}
                label={this.props.intl.formatMessage({id: 'page.body.wallets.tabs.withdraw.content.code2fa'})}
                buttonLabel={this.props.intl.formatMessage({id: 'page.header.signIn'})}
                message={this.props.intl.formatMessage({id: 'page.password2fa.message'})}
                codeFocused={codeFocused}
                otpCode={otpCode}
                error={error2fa}
                handleOtpCodeChange={this.handleChangeOtpCode}
                handleChangeFocusField={this.handle2faFocus}
                handleClose2fa={this.handleClose}
            />
        );
    };

    private refreshError = () => {
        this.setState({
            emailError: '',
            passwordError: '',
        });
    };

    private handleChangeOtpCode = (value: string) => {
        this.setState({
            error2fa: '',
            otpCode: value,
        });
    };

    private handleSignIn = () => {
        const {email, password} = this.state;
        this.props.signIn({
            email,
            password,
        });
    };


    private handleDisable2FA = () => {
        const sessionsConfig: RequestOptions = {
            apiVersion: 'barong',
        };
        API.post(sessionsConfig)('/identity/otp/disable', {
            email: this.state.email,
            password: this.state.password,
        }).then(response => {
            this.props.alertSuccess({
                type: 'success',
                message: [response.message],
            });
            this.handleClose();
        }).catch(error => {
            this.props.alertSuccess({
                type: 'error',
                message: [error.message],
            });
        });
    };

    private handle2FASignIn = () => {
        const {email, password, otpCode} = this.state;
        if (!otpCode) {
            this.setState({
                error2fa: 'Please enter 2fa code',
            });
        } else {
            this.props.signIn({
                email,
                password,
                otp_code: otpCode,
            });
        }
    };

    private handleSignUp = () => {
        this.props.history.push('/signup');
    };

    private forgotPassword = () => {
        this.props.history.push('/forgot_password');
    };

    private handleFieldFocus = (field: string) => {
        switch (field) {
            case 'email':
                this.setState(prev => ({
                    emailFocused: !prev.emailFocused,
                }));
                break;
            case 'password':
                this.setState(prev => ({
                    passwordFocused: !prev.passwordFocused,
                }));
                break;
            default:
                break;
        }
    };

    private handle2faFocus = () => {
        this.setState(prev => ({
            codeFocused: !prev.codeFocused,
        }));
    };

    private validateForm = () => {
        const {email, password} = this.state;
        const isEmailValid = email.match(EMAIL_REGEX);

        if (!isEmailValid) {
            this.setState({
                emailError: this.props.intl.formatMessage({id: ERROR_INVALID_EMAIL}),
                passwordError: '',
            });
            return;
        }
        if (!password) {
            this.setState({
                emailError: '',
                passwordError: this.props.intl.formatMessage({id: ERROR_EMPTY_PASSWORD}),
            });
            return;
        }
    };

    private handleChangeEmailValue = (value: string) => {
        this.setState({
            email: value,
        });
    };

    private handleChangePasswordValue = (value: string) => {
        this.setState({
            password: value,
        });
    };

    private handleClose = () => {
        this.props.signInRequire2FA({require2fa: false});
        this.props.signInShow2FActions({show2faActions: false});
    };
}

const mapStateToProps: MapStateToProps<ReduxProps, {}, RootState> = state => ({
    alert: selectAlertState(state),
    isLoggedIn: selectUserLoggedIn(state),
    loading: selectUserFetching(state),
    require2FA: selectSignInRequire2FA(state),
    show2faActions: selectShow2faActions(state),
    requireEmailVerification: selectSignUpRequireVerification(state),
});

const mapDispatchProps: MapDispatchToPropsFunction<DispatchProps, {}> = dispatch => ({
    signIn: data => dispatch(signIn(data)),
    signInError: error => dispatch(signInError(error)),
    signInRequire2FA: payload => dispatch(signInRequire2FA(payload)),
    signInShow2FActions: payload => dispatch(show2FAActions(payload)),
    signUpRequireVerification: data => dispatch(signUpRequireVerification(data)),
    alertSuccess: payload => dispatch(alertPush(payload)),
});

// tslint:disable no-any
const SignInScreen = injectIntl(
    withRouter(connect(
        mapStateToProps,
        mapDispatchProps,
    )(SignIn) as any),
);
// tslint:enable no-any

export {
    SignInScreen,
};
