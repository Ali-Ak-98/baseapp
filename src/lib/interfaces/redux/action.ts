import { Action } from 'redux';

export interface ActionWith<PayloadType = any> extends Action {
    payload: PayloadType;
}
