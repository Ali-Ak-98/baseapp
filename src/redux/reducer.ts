import { combineReducers } from 'redux';
import { publicReducer, userReducer } from '../modules/app';

export const rootReducer = combineReducers({
    public: publicReducer,
    user: userReducer,
});