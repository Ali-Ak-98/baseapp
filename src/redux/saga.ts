import { all, fork } from 'redux-saga/effects';
import { SagaIterator } from 'redux-saga';

import { localizationSaga } from 'src/units/localization/redux';
import { keepSaga } from 'src/units/keep/redux';
import { commonSaga } from 'src/units/common/redux';

export function* rootSaga(): SagaIterator {
    yield all([fork(localizationSaga)]);
    yield all([fork(keepSaga)]);
    yield all([fork(commonSaga)]);
}
