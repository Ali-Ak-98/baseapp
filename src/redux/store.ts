import { applyMiddleware, compose, createStore } from 'redux';
import sagaMiddlewareFactory from 'redux-saga';
import { rangerSagas } from '../modules/public/ranger';
import { rootReducer } from './reducer';
import { rootSaga } from './saga';

const sagaMiddleware = sagaMiddlewareFactory();
const rengerMiddleware = sagaMiddlewareFactory();

// tslint:disable-next-line:no-any
const composeEnhancer: typeof compose = (window as any)
    .__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
    rootReducer,
    composeEnhancer(
        applyMiddleware(
            sagaMiddleware,
            rengerMiddleware,
        ),
    ),
);

// then run the saga
sagaMiddleware.run(rootSaga);
rengerMiddleware.run(rangerSagas);

export {
    store,
};
