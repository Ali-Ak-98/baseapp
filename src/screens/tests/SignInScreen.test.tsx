import { shallow } from 'enzyme';
import * as React from 'react';
import { InjectedIntlProps } from 'react-intl';
import { connect, Provider } from 'react-redux';
import { store } from '../../redux';
import { SignInScreen } from '../SignInScreen';

const Identity = connect()(SignInScreen);

const setup = (props: Partial<InjectedIntlProps> = {}) =>
    shallow(
        <Provider store={store}>
            <Identity />
        </Provider>,
    );

describe('SignInScreen', () => {
    const wrapper = setup();

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
