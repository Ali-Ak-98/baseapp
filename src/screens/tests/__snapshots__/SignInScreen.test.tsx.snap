// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInScreen should render 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeState": Object {
        "public": Object {
          "alerts": Object {
            "alerts": Array [],
          },
          "blocklistAccess": Object {
            "error": false,
            "loading": false,
            "status": "",
            "success": false,
          },
          "colorTheme": Object {
            "chartRebuild": false,
            "color": "basic",
            "marketSelectorActive": false,
            "sideBarActive": false,
          },
          "configs": Object {
            "data": Object {
              "captcha_type": "none",
              "password_min_entropy": 0,
            },
            "loading": false,
            "success": false,
          },
          "currencies": Object {
            "list": Array [],
            "loading": false,
          },
          "customization": Object {
            "loading": false,
            "success": false,
          },
          "depth": Object {
            "asks": Array [],
            "bids": Array [],
            "loading": false,
          },
          "i18n": Object {
            "lang": "en",
          },
          "incrementDepth": Object {
            "asks": Array [],
            "bids": Array [],
            "loading": false,
            "marketId": undefined,
            "sequence": null,
          },
          "kline": Object {
            "data": Array [],
            "last": undefined,
            "loading": false,
            "marketId": undefined,
            "period": undefined,
            "range": Object {
              "from": 0,
              "to": 0,
            },
          },
          "markets": Object {
            "currentMarket": undefined,
            "list": Array [],
            "loading": false,
            "tickerLoading": false,
            "tickers": Object {},
          },
          "memberLevels": Object {
            "loading": false,
          },
          "orderBook": Object {
            "asks": Array [],
            "bids": Array [],
            "loading": false,
          },
          "ranger": Object {
            "connected": false,
            "subscriptions": Array [],
            "withAuth": false,
          },
          "recentTrades": Object {
            "list": Array [],
            "loading": false,
          },
          "rgl": Object {
            "layouts": Object {
              "lg": Array [
                Object {
                  "h": 21,
                  "i": "1",
                  "isDraggable": false,
                  "isResizable": false,
                  "maxH": 21,
                  "minH": 21,
                  "minW": 4,
                  "w": 4,
                  "x": 20,
                  "y": 0,
                },
                Object {
                  "h": 19.6,
                  "i": "2",
                  "isDraggable": false,
                  "isResizable": false,
                  "minH": 21,
                  "minW": 5,
                  "w": 12,
                  "x": 4,
                  "y": 0,
                },
                Object {
                  "h": 39.7,
                  "i": "3",
                  "isDraggable": false,
                  "isResizable": false,
                  "minH": 21,
                  "minW": 4,
                  "w": 4,
                  "x": 16,
                  "y": 0,
                },
                Object {
                  "h": 9,
                  "i": "4",
                  "isDraggable": false,
                  "isResizable": false,
                  "minH": 9,
                  "minW": 5,
                  "w": 12,
                  "x": 4,
                  "y": 40,
                },
                Object {
                  "h": 11.100000000000001,
                  "i": "5",
                  "isDraggable": false,
                  "isResizable": false,
                  "minH": 7,
                  "minW": 5,
                  "w": 12,
                  "x": 4,
                  "y": 60,
                },
                Object {
                  "h": 39.7,
                  "i": "6",
                  "isDraggable": false,
                  "isResizable": false,
                  "minH": 10,
                  "minW": 4,
                  "w": 4,
                  "x": 0,
                  "y": 0,
                },
                Object {
                  "h": 18.7,
                  "i": "7",
                  "isDraggable": false,
                  "isResizable": false,
                  "minH": 10,
                  "minW": 4,
                  "w": 4,
                  "x": 20,
                  "y": 40,
                },
              ],
              "md": Array [
                Object {
                  "h": 21,
                  "i": "1",
                  "isDraggable": false,
                  "isResizable": false,
                  "maxH": 21,
                  "minH": 21,
                  "minW": 4,
                  "w": 8,
                  "x": 16,
                  "y": 18,
                },
                Object {
                  "h": 19.6,
                  "i": "2",
                  "isDraggable": false,
                  "isResizable": false,
                  "minH": 21,
                  "minW": 5,
                  "w": 16,
                  "x": 0,
                  "y": 0,
                },
                Object {
                  "h": 39.7,
                  "i": "3",
                  "isDraggable": false,
                  "isResizable": false,
                  "minH": 21,
                  "minW": 4,
                  "w": 8,
                  "x": 16,
                  "y": 0,
                },
                Object {
                  "h": 9,
                  "i": "4",
                  "isDraggable": false,
                  "isResizable": false,
                  "minH": 9,
                  "minW": 5,
                  "w": 16,
                  "x": 0,
                  "y": 40,
                },
                Object {
                  "h": 11.100000000000001,
                  "i": "5",
                  "isDraggable": false,
                  "isResizable": false,
                  "minH": 7,
                  "minW": 5,
                  "w": 16,
                  "x": 0,
                  "y": 60,
                },
                Object {
                  "h": 39.7,
                  "i": "6",
                  "isDraggable": false,
                  "isResizable": false,
                  "minH": 10,
                  "minW": 4,
                  "w": 16,
                  "x": 0,
                  "y": 60,
                },
                Object {
                  "h": 18.7,
                  "i": "7",
                  "isDraggable": false,
                  "isResizable": false,
                  "minH": 10,
                  "minW": 4,
                  "w": 8,
                  "x": 20,
                  "y": 36,
                },
              ],
              "sm": Array [
                Object {
                  "h": 22,
                  "i": "1",
                  "isDraggable": false,
                  "maxH": 22,
                  "minH": 22,
                  "minW": 5,
                  "w": 12,
                  "x": 0,
                  "y": 12,
                },
                Object {
                  "h": 30,
                  "i": "2",
                  "isDraggable": false,
                  "minH": 30,
                  "minW": 5,
                  "w": 12,
                  "x": 0,
                  "y": 28,
                },
                Object {
                  "h": 18,
                  "i": "3",
                  "isDraggable": false,
                  "minH": 12,
                  "minW": 3,
                  "w": 12,
                  "x": 0,
                  "y": 58,
                },
                Object {
                  "h": 12,
                  "i": "4",
                  "isDraggable": false,
                  "minH": 12,
                  "minW": 7,
                  "w": 12,
                  "x": 0,
                  "y": 94,
                },
                Object {
                  "h": 20,
                  "i": "5",
                  "isDraggable": false,
                  "minH": 12,
                  "minW": 7,
                  "w": 12,
                  "x": 0,
                  "y": 82,
                },
                Object {
                  "h": 20,
                  "i": "6",
                  "isDraggable": false,
                  "minH": 12,
                  "minW": 7,
                  "w": 12,
                  "x": 0,
                  "y": 106,
                },
                Object {
                  "h": 20,
                  "i": "7",
                  "isDraggable": false,
                  "minH": 12,
                  "minW": 7,
                  "w": 12,
                  "x": 0,
                  "y": 126,
                },
              ],
            },
          },
        },
        "user": Object {
          "addresses": Object {
            "loading": false,
          },
          "apiKeys": Object {
            "apiKeys": Array [],
            "dataLoaded": false,
            "modal": Object {
              "active": false,
            },
          },
          "auth": Object {
            "current_password_entropy": 0,
            "emailVerified": false,
            "require2FA": false,
            "requireVerification": false,
          },
          "beneficiaries": Object {
            "activate": Object {
              "data": Object {
                "currency": "",
                "data": Object {
                  "address": "",
                },
                "id": 0,
                "name": "",
                "state": "",
              },
              "fetching": false,
              "success": false,
            },
            "create": Object {
              "data": Object {
                "currency": "",
                "data": Object {
                  "address": "",
                },
                "id": 0,
                "name": "",
                "state": "",
              },
              "fetching": false,
              "success": false,
            },
            "delete": Object {
              "data": Object {
                "id": 0,
              },
              "fetching": false,
              "success": false,
            },
            "fetch": Object {
              "data": Array [],
              "fetching": false,
              "success": false,
            },
          },
          "captchaKeys": Object {
            "loading": false,
            "success": false,
          },
          "customizationUpdate": Object {
            "loading": false,
            "success": false,
          },
          "documents": Object {
            "loading": false,
          },
          "history": Object {
            "fetching": false,
            "list": Array [],
            "nextPageExists": false,
            "page": 0,
          },
          "identity": Object {
            "edit": Object {},
            "send": Object {},
          },
          "label": Object {
            "data": Array [],
            "isFetching": false,
          },
          "newHistory": Object {
            "fetching": false,
            "list": Array [],
          },
          "openOrders": Object {
            "cancelError": false,
            "cancelFetching": false,
            "fetching": false,
            "list": Array [],
          },
          "orders": Object {
            "amount": "",
            "currentPrice": undefined,
            "executeLoading": false,
            "orderType": "",
          },
          "ordersHistory": Object {
            "cancelAllError": false,
            "cancelAllFetching": false,
            "cancelError": false,
            "cancelFetching": false,
            "fetching": false,
            "list": Array [],
            "nextPageExists": false,
            "pageIndex": 0,
          },
          "password": Object {
            "forgotPasswordChanged": false,
            "forgotPasswordRequested": false,
            "loading": false,
          },
          "phone": Object {
            "codeSend": false,
          },
          "profile": Object {
            "passwordChange": Object {
              "success": false,
            },
            "twoFactorAuth": Object {
              "barcode": "",
              "url": "",
            },
            "userData": Object {
              "isFetching": false,
              "user": Object {
                "created_at": "",
                "email": "",
                "labels": Array [],
                "level": 0,
                "otp": false,
                "phone": Array [],
                "profiles": Array [],
                "referal_uid": "",
                "role": "",
                "state": "",
                "uid": "",
                "updated_at": "",
              },
            },
          },
          "sendEmailVerification": Object {
            "loading": false,
            "success": false,
          },
          "userActivity": Object {
            "list": Array [],
            "loading": false,
            "page": 0,
            "total": 0,
          },
          "wallets": Object {
            "wallets": Object {
              "list": Array [],
              "loading": false,
              "mobileWalletChosen": "",
              "selectedWalletAddress": "",
              "selectedWalletCurrency": "",
              "withdrawSuccess": false,
            },
          },
          "withdrawLimit": Object {
            "data": Object {
              "currency": "",
              "limit": "0.0",
              "period": 0,
              "withdrawal_amount": "0.0",
            },
            "loading": false,
            "success": false,
          },
        },
      },
    }
  }
>
  <Connect(InjectIntl(withRouter(Connect(SignIn)))) />
</ContextProvider>
`;
